tests/virt_autotest/reset_partition.pm

```diff
@@ -0,0 +1,107 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package reset_partition;
+# Summary: reset_partition: Ensure and reset hard disk partition for virt_autotest.
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use virt_utils;
+use ipmi_backend_utils;
+use Utils::Architectures;
+use utils qw(zypper_call);
+use version_utils 'is_sle';
+use virt_autotest::utils qw(is_xen_host);
+
+sub reset_partition {
+    my $libvirt_disk = shift;
+    my $new_disk     = "";
+    my $virt_point   = "/var/lib/libvirt/";
+
+    # Install requested pkgs
+    my @pkgs = qw/xfsprogs coreutils util-linux/;
+    zypper_call "in @pkgs";
+
+    # Get some debug info about hard disk topology
+    record_info("Report file system disk space", script_output("df"));
+    record_info("Find mounted filesystem",       script_output("findmnt"));
+
+    my @disks = qw/sda sdb sdc/;
+    my $dev   = "/dev/";
+    foreach my $disk (@disks) {
+        assert_script_run('file ' . $dev . $disk);
+    }
+
+    # Assign a New FULL Disk for /var/lib/libvirt
+    if ($libvirt_disk eq 'sda') {
+        $new_disk = "/dev/sdb";
+    } else {
+        $new_disk = "/dev/sda";
```

Comment:
Here is not compatible with your description. Generally speaking, /dev/sda can not be the new full disk. You should choose sdc if sdc exists

Reply:
Good point. 
Here, one point need to be mentioned, refer to the current SLE15+ host installation on X86_64.  root system dir(/) and libvirt dir had deployed at the same disk. means that the current existed hard disks - sda, sdb and sdc on each OSD worker|SUT was able to deployed SLE15+ dynamically , mounted as  root system dir(/) and libvirt dir had deployed at the same time.
sure, refer to the current if..else is very reckless, but which was a simple way here.

Reply:
Your explanation does not explain my concern. I mean you'd better detect unused disk firstly, then pick one from them as your new full disk. That is it.

Reply:
Your sdc seems useless. If you loop over sdc, you should also include it as your choice of new full disk.  But I still think you should detect unused disk firstly then pick one from them. That is it. If you implement this way, you can even save your "loop over all disks" part. IMHO.

Reply:
sure, had updated to use with "loop over all disks" , instead of sd[x] way now.